### PR TITLE
Reorganize test requirements

### PR DESCRIPTION
**Pull request by ssbarnea**
Move test requirements out of tox.ini

### CODE REVIEW

File: test-requirements.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env pip-compile -q --output-file=test-requirements.txt
+# Do not use  --require-hashes because it is incompatible with runtime requirements.
```
Comment (webknjaz) [inline test-requirements.in:2]: Why is that?

File: test-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+py==1.9.0                 # via pytest, pytest-forked
+pyparsing==2.4.7          # via packaging
+pytest-cov==2.10.1        # via -r ./test-requirements.in
+pytest-forked==1.3.0      # via pytest-xdist
+pytest-xdist==2.1.0       # via -r ./test-requirements.in
+pytest==6.0.1             # via -r ./test-requirements.in, pytest-cov, pytest-forked, pytest-xdist
+six==1.15.0               # via packaging
+toml==0.10.1              # via pytest
+
+# The following packages are considered to be unsafe in a requirements file:
```
Comment (webknjaz) [inline test-requirements.txt:24]: You forgot `--allow-unsafe`

File: tox.ini [hunk 3]
```diff
@@ -20,15 +20,7 @@ deps =
...
-  ruamel.yaml==0.16.5  # NOTE: 0.15.34 has issues with py37
-  flake8
-  pep8-naming
-  pytest
-  pytest-cov
-  pytest-xdist
-  # Needed to avoid DeprecationWarning errors in pytest:
-  setuptools >= 40.4.3
-  wheel
+  -r test-requirements.txt
```
Comment (webknjaz) [inline tox.ini:23]:
```suggestion
  -r test-requirements.in
  -c test-requirements.txt
```

**Review by webknjaz — CHANGES_REQUESTED** [review]
This missing hashes and dependabot config entry.

P.S. Since you don't like files in root, feel free to move them to a requirements folder following pypa/warehouse example.

File: test-requirements.in [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env pip-compile -q --allow-unsafe --output-file=test-requirements.txt
+# Avoid using --generate-hashes as it breaks pip install from tox with:
+# ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
```
Comment (webknjaz) [inline test-requirements.in:3]: Uh, I know what to do but that's for another PR

Comment (ssbarnea) [reply test-requirements.in:3]: That's the spirit, lets progress even if some parts may not be ideal. As long there is no clear regression in a change, there is no need to delay it even further, especially for long standing ones.

**Review by webknjaz — APPROVED** [review]

**Pull request merged**